src/main/java/net/dv8tion/jda/core/handle/PresenceUpdateHandler.java

```diff
@@ -158,7 +168,32 @@ protected Long handleInternally(JSONObject content)
             else
             {
                 //In this case, this PRESENCE_UPDATE is for a Relation.
+                if (api.getAccountType() != AccountType.CLIENT)
+                    return null;
+                JDAClient client = api.asClient();
+                FriendImpl friend = (FriendImpl) client.getFriendById(userId);
 
+                if (friend != null)
+                {
+                    if (!friend.getOnlineStatus().equals(status))
+                    {
+                        OnlineStatus oldStatus = friend.getOnlineStatus();
+                        friend.setOnlineStatus(status);
+                        api.getEventManager().handle(
+                            new UserOnlineStatusUpdateEvent(
+                                api, responseNumber,
+                                user, null, oldStatus));
+                    }
+                    if (!Objects.equals(friend.getGame(), nextGame))
+                    {
+                        Game oldGame = friend.getGame();
+                        friend.setGame(nextGame);
+                        api.getEventManager().handle(
+                            new UserGameUpdateEvent(
+                                api, responseNumber,
+                                user, null, oldGame));
```

Comment:
We should be firing FriendXEvents, not UserXEvents I think.
In that same vain though, we probably should rename the User events to Member events?

Reply:
Or, at a minimum, take a Friend object to these User events so that you can call `getFriend` like you can call `getMember`.

Reply:
That is already supported.